class.jetpack.php

```diff
@@ -4601,6 +4601,8 @@ function build_connect_url( $raw = false, $redirect = false, $from = false ) {
 
 			$redirect = $redirect ? esc_url_raw( $redirect ) : esc_url_raw( menu_page_url( 'jetpack', false ) );
 
+			$gp_locale = GP_Locales::by_field( 'wp_locale', get_locale() );
```

Comment:
if you search wpcom's code for `jp_lang`, you'll see how I'm handling this for converting jetpack's locale to wpcom